addons/auth_signup/models/res_partner.py

```diff
@@ -134,24 +102,19 @@ def signup_get_auth_param(self):
             partner = partner.sudo()
             if allow_signup and not partner.user_ids:
                 partner.signup_prepare()
-                res[partner.id]['auth_signup_token'] = partner.signup_token
+                res[partner.id]['auth_signup_token'] = partner._generate_signup_token()
             elif partner.user_ids:
                 res[partner.id]['auth_login'] = partner.user_ids[0].login
         return res
 
     def signup_cancel(self):
-        return self.write({'signup_token': False, 'signup_type': False, 'signup_expiration': False})
+        return self.write({'signup_type': None})
 
-    def signup_prepare(self, signup_type="signup", expiration=False):
+    def signup_prepare(self, signup_type="signup"):
```

Comment:
Can be done as a second iteration / cleaning pull request, but all these `def signup_` can be privatized.